### PR TITLE
Refactor/issue 656 dbreeze Part 1

### PR DESCRIPTION
**Pull request by Aprogiena**
This starts the process of fixing DBreeze usage as discussed in #656. We separate serializer to its own class and we try to get rid of DBreezeSingleThreadSession. In this PR we start with ChainRepository.

### CODE REVIEW

**Comment by bokobza** [comment]
what's wrong with this @Aprogiena ? 😢

**Comment by Aprogiena** [comment]
It is OK now :-) There was a PeriodicTask, which we wanted to replace with AsyncLoop and that caused the chain not to be saved properly. Now it is OK, tested!

**Comment by Aprogiena** [comment]
Hmm looks like tests are not passing here, damn.

**Comment by Aprogiena** [comment]
works now

File: Stratis.Bitcoin/Base/ChainRepository.cs [hunk 1]
```diff
@@ -5,90 +5,106 @@
...
+                        {
+                            first = false;
+                            Guard.Assert(tip.HashBlock == chain.Genesis.HashBlock); // can't swap networks
+                        }
                     }
+
+                    if (tip == null)
+                        return;
+
+                    this.locator = tip.GetLocator();
```
Comment (dangershony) [inline Stratis.Bitcoin/Base/ChainRepository.cs]: Is this required?
`this.locator = tip.GetLocator();`

Comment (dangershony) [reply Stratis.Bitcoin/Base/ChainRepository.cs]: Ah yes ignore

File: Stratis.Bitcoin/Utilities/DBreezeSerializer.cs [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    /// Implementation of serialization and deserialization of objects that go into the DBreeze database.
+    /// </summary>
+    public class DBreezeSerializer
+    {
+        /// <summary>
+        /// Initializes custom serializers for DBreeze engine.
+        /// </summary>
+        public void Initialize()
+        {
+            CustomSerializator.ByteArraySerializator = Serializer;
```
Comment (dangershony) [inline Stratis.Bitcoin/Utilities/DBreezeSerializer.cs]: `this.Serializer`

Comment (Aprogiena) [reply Stratis.Bitcoin/Utilities/DBreezeSerializer.cs]: damn, always at least one instance i forget :) fixed

**Pull request merged**